Дополнена функциональность по финальному ТЗ спринта 6

myakunovre opened the pull request:
1. Исключить дублирование задач в истории
2. Историю выводить в порядке обращения к задачам
3. Хранение задач в истории реализовать через HashMap и связанный список.
4. Обновить unit-тесты

myakunovre committed:
1. Исключить дублирование задач в истории
2. Историю выводить в порядке обращения к задачам
3. Хранение задач в истории реализовать через HashMap и связанный список.
4. Обновить unit-тесты

myakunovre committed:
Изменить имя параметра в параметризированном классе TaskLinkedList

myakunovre committed:
Изменить имя параметра в параметризированном классе Node

VladimirTitov4 commented on src/manager/TaskLinkedList.java:
```diff
@@ -0,0 +1,53 @@
+        final Node<T> oldLast = last;
+        final Node<T> newNode = new Node<>(oldLast, task, null);
+        last = newNode;
+        if (oldLast == null)
```
Ставьте фигурные скобки всегда

myakunovre replied:
ок, исправлено

VladimirTitov4 commented on src/manager/TaskLinkedList.java:
```diff
@@ -0,0 +1,53 @@
+        size++;
+    }
+
+    List<T> getTasks() {
```
Модификаторы доступа нужно добавить

myakunovre replied:
1. У метода getTask() добавил.
2. Переменную size исключил за ненадобностью (сначала поставил private, после чего идея предложила исключить ее вообще, т.к. size менялся только в методах этого класса и больше нигде не использовался).

myakunovre committed:
1. В методе linkLast класса TaskLinkedList добавлены фигурные скобки (по замечанию ревью в github).
2. В классе TaskLinkedList исключено поле size, т.к. оно не используется в программе в целом.
3. Исключены комментарии в коде (по замечанию ревью в Практикуме)

myakunovre commented:
Ревьюер принял, делаю merge
